blackmarket/Managers/SessionManager.swift

```diff
@@ -0,0 +1,57 @@
+//
+//  SessionDataManager.swift
+//  ios-base
+//
+//  Created by Juan Pablo Mazza on 11/8/16.
+//  Copyright © 2016 Rootstrap Inc. All rights reserved.
+//
+
+import UIKit
+import Alamofire
+import Combine
+
+internal class SessionManager: CurrentUserSessionProvider {
+    
+    static let Authenticated = PassthroughSubject<Bool, Never>()
```

Comment:
This should be a read-only variable, a Publisher would make more sense here:

```suggestion
    var isAuthenticatedSubject: AnyPublisher<Bool, Never>() {
       // Decide from session object in the user defaults. Pls see my other comments. 
    }
```